app/helpers/address_helper.rb

```diff
@@ -123,4 +123,19 @@ def compare(ordnance_address, edited_address)
 
     sanitized_edited_address & sanitized_ordnance_address == sanitized_ordnance_address
   end
+
+  def remove_selected_address_changes(selected_address)
+    address_hash = JSON.parse(selected_address).to_h
+    selected_address_keys = address_hash.keys
+
+    address_to_return = address_hash.select do |key, value|
+      WANTED_VALUES.include?(key) && value.instance_of?(String)
+    end
+
+    unless address_to_return.keys == selected_address_keys
+      address_to_return.delete("UPRN")
+    end
```

Comment:
I don't think this block is necessary, since non-malicious users won't modify the address, and therefore the UPRN doesn't matter for these users.

Reply:
Thanks @bilbof - I have removed this block as requested.  However, personally I disagree - it could be seen that if we have a valid UPRN then the rest of the address can reasonably be ignored and the entry assumed to be from a real user.  Removing the UPRN - when we know that someone has munged the address - is a safe-guard against this assumption.